_api/hasura/cron/giveOnchainSyncer.ts

```diff
@@ -0,0 +1,79 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { attestGiveOnchain } from '../../../api-lib/eas';
+import { order_by } from '../../../api-lib/gql/__generated__/zeus';
+import { adminClient } from '../../../api-lib/gql/adminClient';
+import { verifyHasuraRequestMiddleware } from '../../../api-lib/validate';
+
+const LIMIT = 10;
+
+async function handler(_req: VercelRequest, res: VercelResponse) {
+  const gives = await giveToSync();
+
+  try {
+    let errors = 0;
+    let success = 0;
+    for (const give of gives) {
+      try {
+        await attestGiveOnchain(give);
+        success++;
+      } catch (e: any) {
+        errors++;
+      }
+    }
+
+    // eslint-disable-next-line no-console
+    console.log('Syncing complete', { success, errors });
+    res.status(200).json({
+      success: true,
+      gives: gives.length,
+      errorsCount: errors,
+      successCount: success,
+    });
+  } catch (e: any) {
```

Comment:
will this "try ..catch" catch any errors? don't we need to throw again if errors is greater than 0

Reply:
good catch - moved the give fetching into the block